core/frontend/src/components/parameter-editor/ParameterEditor.vue

```diff
@@ -0,0 +1,205 @@
+                  label="Value"
+                  type="number"
+                  @change="saveParam"
+                />
+              </v-col>
+            </v-row>
+          </v-container>
+        </v-card-text>
+
+        <v-card-actions>
+          <v-spacer />
+          <v-btn
+            color="blue darken-1"
+            text
+            @click="editDialog = false"
+          >
+            Cancel
+          </v-btn>
+          <v-btn
+            color="blue darken-1"
+            text
+            @click="saveParam"
+          >
+            Save
+          </v-btn>
+        </v-card-actions>
+      </v-card>
+    </v-dialog>
+  </v-card>
+</template>
+
+<script lang="ts">
+import Vue from 'vue'
+
+// import SpinningLogo from '../common/SpinningLogo.vue'
+import mavlink2rest from '@/libs/MAVLink2Rest'
+import autopilot_data from '@/store/autopilot'
+import Parameter from '@/types/autopilot/parameter'
+
+export default Vue.extend({
+  name: 'ParameterEditor',
+  data() {
+    return {
+      search: '',
+      editedParam: null as (null | Parameter),
+      editDialog: false,
+      editedBitmask: [] as number[],
+    }
+  },
+  computed: {
+    params() {
+      return autopilot_data.parameters
+    },
+    asComboBoxItems() {
+      const entries = Object.entries(this.editedParam?.options ?? [])
+      return entries.map(([value, name]) => ({ text: name, value: parseInt(value, 10), disabled: false }))
+    },
+  },
+  methods: {
+    editParam(param: Parameter) {
+      if (param.readonly) {
+        return
+      }
+      this.editedParam = param
+      this.editDialog = true
+      this.editedBitmask = []
+      if (param.bitmask === undefined) {
+        console.log('no bitmask!')
+        return
+      }
+      let i = 0
+      for (i = 0; i < 32; i += 1) {
+        // eslint-disable-next-line no-bitwise
+        if (2 ** i & param.value) {
+          this.editedBitmask.push(2 ** i)
+        }
+      }
+    },
+    saveParam() {
+      this.editDialog = false
+      if (this.editedParam == null) {
+        return
+      }
+      if (this.editedParam.bitmask !== undefined) {
+        this.editedParam.value = this.editedBitmask.reduce((accumulator, current) => accumulator + current, 0)
+      }
+      let value = 0
+      if (typeof this.editedParam.value === 'string') {
+        value = parseFloat(this.editedParam.value)
+      } else {
+        value = this.editedParam.value
+      }
+      const param_id = [...this.editedParam.name]
+      while (param_id.length < 16) {
+        param_id.push('\0')
+      }
+      mavlink2rest.sendMessage({
+        header: {
+          system_id: 255,
+          component_id: 0,
+          sequence: 0,
+        },
+        message: {
+          type: 'PARAM_SET',
+          param_value: value,
+          target_system: 0,
+          target_component: 0,
+          param_id,
+          param_type: {
+            type: 'MAV_PARAM_TYPE_UINT8',
```

Comment:
Wait, but this will not work with anything besides `uint8`, right ? What about real32 and the others ?

Reply:
Ardupilot just ignores the type in the message as it knows the types already. Plus this info isnt in the metadata?

Reply:
> Ardupilot just ignores the type in the message as it knows the types already.

cool



> Plus this info isnt in the metadata?

IIRC nops